workers/shared-worker-parse-error-failure.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<title>SharedWorker: parse error failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+promise_test(async () => {
+  const scriptURL = 'modules/resources/syntax-error.js';
+  const worker = new SharedWorker(scriptURL, { name: 'classic',
+                                               type: 'classic' });
+  return new Promise(resolve => worker.onerror = resolve);
```

Comment:
It would be good to test that the value passed to onerror is an ErrorEvent. Also, worth testing that there are no other arguments, as `window.onerror` gets extra arguments but `worker.onerror` should not.

Maybe that is scope creep for this test... But if we don't want to test those things, then we should change the test description to say `error event` instead of `ErrorEvent`.

Reply:
I think the event we get here is not defined as ErrorEvent in the spec but Event or ErrorEvent.
https://html.spec.whatwg.org/multipage/indices.html#event-error
I seems better to modify the description as you said.

Reply:
FYI: FireFox returns Event here.
https://bugzilla.mozilla.org/show_bug.cgi?id=1610438

Reply:
Event is correct. The spec says "fire an event" with no constructor arg, and step 1 of https://dom.spec.whatwg.org/#concept-event-fire says to use `Event`.

The index is a non-normative summary of every event named `error` in the entire HTML spec. You have to look at individual "fire an event" sites to see which constructor they use.

Reply:
Upon reviewing the spec PR I think it's actually important that we test the type (and arguments) here. This is test of new functionality and it's easy to get it wrong. We should be sure that it gets test coverage.

Reply:
I added the test to check if the event's type is 'error'.

Reply:
Sorry, by "type" I actually meant "constructor". Please test its `.constructor` property, and also check the number of arguments sent to `onerror`.

Reply:
Sorry for taking it in the wrong way. I modified as you suggested.
I found the test to check onerror for dedicated workers:
https://github.com/web-platform-tests/wpt/blob/master/workers/constructors/Worker/AbstractWorker.onerror.html
and it seems that this test should be also merged into wpt/workers/modules/dedicated-worker-parse-error-failure.html so that we can test parse error occurred in statically imported scripts (but not in this CL).